packages/client/src/index.ts

```diff
@@ -249,6 +249,8 @@ async function maybeHandleError (resPromise: Promise<globalThis.Response>): Prom
   const err = new Error(`unexpected status: ${res.status}`)
   try {
     Object.assign(err, await res.json())
-  } catch {}
+  } catch {
+    throw err
+  }
   throw err
```

Comment:
The `catch` scenario will still do `throw err` anyway, right? So the behaviour here doesn't actually change.

Reply:
Yeah I could possibly just wrap this all in a `finally`

Reply:
Ah, an excellent use of the old `finally` statement.